diff --git a/modules/highgui/CMakeLists.txt b/modules/highgui/CMakeLists.txt
index ae9c49c1cf..6fafa33fda 100644
--- a/modules/highgui/CMakeLists.txt
+++ b/modules/highgui/CMakeLists.txt
@@ -125,8 +125,6 @@ elseif(HAVE_QT)
     endif()
 
     foreach(dt_dep ${qt_deps})
-      link_libraries(${Qt${QT_VERSION_MAJOR}${dt_dep}})
-      include_directories(${Qt${QT_VERSION_MAJOR}${dt_dep}_INCLUDE_DIRS})
       list(APPEND HIGHGUI_LIBRARIES ${Qt${QT_VERSION_MAJOR}${dt_dep}_LIBRARIES})
     endforeach()
   else()
